vite.plugins.ts

```diff
@@ -0,0 +1,94 @@
+import { render } from "@lit-labs/ssr/lib/render-lit-html";
+import { readFileSync } from "fs";
+import { PluginOption } from "vite";
+import { pageContent as aboutStaticContent } from "./src/frontend/src/flows/about";
+import { I18n } from "./src/frontend/src/i18n";
+
+/**
+ * Read the II canister ID from dfx's local state
+ */
+const readCanisterId = (): string => {
+  const canisterIdsJsonFile = "./.dfx/local/canister_ids.json";
+
+  try {
+    const {
+      internet_identity: { local: canisterId },
+    } = JSON.parse(readFileSync(canisterIdsJsonFile, "utf-8"));
+
+    const assertNonNullish: (
+      value: string
+    ) => asserts value is NonNullable<string> = (value: string): void => {
+      if (value === null || value === undefined) {
+        throw new Error("Internet identity canister ID undefined");
+      }
+    };
```

Comment:
I think config files really benefit from being concise, even if they're not super robust (i.e. it's ok if they break on bad input). Would it make sense to use something simpler and keep the config short? Any problem with the original code?

Reply:
Once `@dfinity/utils` is added I can remove this piece of code and use the utils. It will become shorter that way.

Reply:
Ok, worth using the original code until then? Or did the code not work with vite anymore?

Reply:
`@dfinity/utils` is now in so I was able to use its assertion utils.